File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.py
import math

def checkPrimeNumber(x):
    # 2부터 x의 제곱근까지의 숫자
    for i in range(2, int(math.sqrt(x) + 1)):
    	if x % i == 0: # 나눠떨어지는 숫자가 있으면 소수가 아님
        	return False
        
    return True	
    
def solution(n):
    answer = 0
    
    for i in range(2, n+1):
        if checkPrimeNumber(i):
            answer += 1
            
    return answer

File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.cpp
#include <string>
#include <vector>
#include <cmath>

using namespace std;

bool checkPrimeNumber(int n){
    int root_n = static_cast<int>(sqrt(n));
    
    for(int i = 2; i <= root_n; i++){
        if(n % i == 0){
            return false;
        }
    }
    return true;
}

int solution(int n) {
    int answer = 0;
    
    for(int i = 2; i <= n; i++){
        if(checkPrimeNumber(i)) answer++;
    }
    
    return answer;
}